feat(experiments): support max, min and avg. aggregations

andehen opened the pull request:
## Problem
We currently only support `sum` as aggregation type per entity.

## Changes
Add support for `min`, `max` and `avg`.
Fixes https://github.com/PostHog/posthog/issues/33299. Was requested recently in a support ticket.

![Screenshot 2025-06-06 at 15 18 54](https://github.com/user-attachments/assets/020bcf81-9ffa-4f5c-9ef2-b84de31be78c)

## How did you test this code?

- manually created an experiment and chose the new aggregation types, and verified the results
- added tests for each of the new aggregation types

andehen committed:
WIP first pass

andehen committed:
extract is_continous to function

andehen committed:
add more tests

andehen committed:
run schema build

github-actions[bot] committed:
Update query snapshots

rodrigoi commented:
do we need to update `metricToQuery`?

https://github.com/PostHog/posthog/blob/d17bec6843f770026c4286e90656b2e779a71231/frontend/src/scenes/experiments/utils.ts#L707-L713

andehen committed:
handle all math types in metricToQuery

andehen commented:
> do we need to update `metricToQuery`?
> 
> https://github.com/PostHog/posthog/blob/d17bec6843f770026c4286e90656b2e779a71231/frontend/src/scenes/experiments/utils.ts#L707-L713

Good catch! Fixed in cf6142e0099b10f063171a99d149f4d2f977d2cc
Added a test for it, and simplified the logic in `metricToQuery` to handle all cases.

jurajmajerik reviewed (approved):
Nice!
